models/model.js

```diff
@@ -23,18 +26,36 @@ exports.selectArticles = async (sort_by) => {
 		'comment_count',
 	];
 
+	const queryArr = [];
+
+	let queryStr = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, CAST(COUNT(comment_id) AS INT) AS comment_count
+				 	FROM articles 
+				 	LEFT JOIN comments ON comments.article_id = articles.article_id`;
+
+	if (topic) {
+		queryStr += ' WHERE topic = $1';
+		queryArr.push(topic);
+	}
+
+	queryStr += ` GROUP BY articles.article_id`;
+
 	if (validSortBys.includes(sort_by)) {
-		const articles = await db.query(
-			`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, CAST(COUNT(comment_id) AS INT) AS comment_count
-			 FROM articles 
-			 LEFT JOIN comments ON comments.article_id = articles.article_id
-			 GROUP BY articles.article_id
-			 ORDER BY ${sort_by} DESC;`
-		);
-		return articles.rows;
+		if (validOrder.includes(order.toUpperCase())) {
+			queryStr += ` ORDER BY ${sort_by} ${order}`;
+		} else {
+			return Promise.reject({ status: 400, msg: 'Invalid order' });
+		}
 	} else {
-		return Promise.reject({status : 400, msg : 'Invalid sort_by'})
+		return Promise.reject({ status: 400, msg: 'Invalid sort_by' });
+	}
+
+	const { rows } = await db.query(queryStr + ';', queryArr);
```

Comment:
The GROUP BY must always come at the end of the query - you'd be free to hardcode the `;` at the end of that concatenation